tests/TestCase/ORM/TableTest.php

```diff
@@ -4515,6 +4516,349 @@ public function testReplacelinksBelongsToManyWithJoint()
     }
 
     /**
+     * Tests that options are being passed through to the internal table method calls.
+     *
+     * @return void
+     */
+    public function testOptionsBeingPassedToImplicitBelongsToManyDeletesUsingSaveReplace()
+    {
+        $articles = TableRegistry::get('Articles');
+
+        $tags = $articles->belongsToMany('Tags');
+        $tags->saveStrategy(BelongsToMany::SAVE_REPLACE);
+        $tags->dependent(true);
+        $tags->cascadeCallbacks(true);
+
+        $actualOptions = null;
+        $tags->junction()->eventManager()->on(
+            'Model.beforeDelete',
+            function (Event $event, Entity $entity, ArrayObject $options) use (&$actualOptions) {
+                $actualOptions = $options->getArrayCopy();
```

Comment:
@markstory Now I'm torn between you and @lorenzo. Personally I tend to be a little overcautious, just to be on the safe side. I have to admit tho, that I currently can't think of reasons for events not being dispatched that aren't covered elsewhere already.

Please tell me one more time, then I can safely blame it on someone else later on in case something goes wrong :wink: